### ISSUE TITLE
Should we add warmup and more iterations to benchmark timing?

### ISSUE THREAD

**Issue by xwang233** [pytorch/pytorch#93634]
In benchmark/common.py, there is this `timed` function https://github.com/pytorch/torchdynamo/blob/b36e750dad8af509a003a386ddcbd2d0d331c983/benchmarks/common.py#L110-L119

, and this function is repeated for 30 times in a benchmark testing. This way of timing basically adds a cuda synchronization at every step.

The argument `times` is not used in function calls in `speedup_experiment`, so it's always the default value of 1. Should we add warmup and more iterations to it? For example,

```python
def timed(model, model_iter_fn, example_inputs, bench_times=20, warmup_times=10, return_result=False):
    synchronize()
    torch.manual_seed(1337)
    # Dont collect outputs to correctly measure timing

    for _ in range(warmup_times):
        model_iter_fn(model, example_inputs, collect_outputs=False)

    synchronize()
    t0 = time.perf_counter()

    for _ in range(bench_times):
        result = model_iter_fn(model, example_inputs, collect_outputs=False)

    synchronize()
    t1 = time.perf_counter()

    time_ret = (t1 - t0) / bench_times

    return (time_ret, result) if return_result else time_ret
```

In a real example of pytorch_struct model, I can add a print statement here 
(note that the printed value is essentially training throughput, larger number means better performance)

```python
print('\n', current_batch_size / timings)
```

https://github.com/pytorch/torchdynamo/blob/b36e750dad8af509a003a386ddcbd2d0d331c983/benchmarks/common.py#L358-L359

and run
```
python benchmarks/torchbench.py --training -d cuda --fast --accuracy-aot-ts-mincut --nvfuser --skip-accuracy-check -k pytorch_struct --generate-aot-autograd-stats --amp --channels-last --repeat 5
```

With the default `times=1` option, I got
```
cuda train pytorch_struct                     
 [[35017.5871647  37323.74538251]
 [35866.07926572 37512.95098422]
 [36008.11497466 37524.31713086]
 [36006.09849868 37431.42716007]
 [35736.21741365 37378.06821014]]
1.044x p=0.00
speedup    gmean=1.04x mean=1.04x
```

With 10 warmup and 20 benchmark iterations, I got
```
cuda train pytorch_struct                     
 [[39687.41066568 51326.46516663]
 [40082.51831748 51371.04681834]
 [40539.70582466 51355.72488685]
 [40619.38590889 51364.33313436]
 [40517.7146401  51370.72731337]]
1.268x p=0.00
speedup    gmean=1.27x mean=1.27x
```

By using the code with new "bench_times" and "warmup_times" arguments, we can still use values 1 and 0 to get the old behavior for backward consistency if needed.

Versions:
- V100, cuda 11.7
- torch_commit: https://github.com/pytorch/pytorch/commit/ce7177f88a8c76351087bd06520681e60591ff50
- torchbench_commit: https://github.com/pytorch/benchmark/commit/98a64dcc527426f3dd1e9179091baf47d2c581f6
- torchdynamo_commit: https://github.com/pytorch/torchdynamo/commit/f2945ab324ef2f5be03dd5ebba32fccb8c7b620f
- torchvision_commit: https://github.com/pytorch/vision/commit/c3dc25528c3bb3b9860773fd8b2d4b47984f0b2d

Related:
https://github.com/pytorch/pytorch/issues/93767

cc @ngimel @jansel @csarofeen @jjsjann123 @kevinstephano @ptrblck
